rustup-init.sh

```diff
@@ -316,20 +317,20 @@ get_architecture() {
     # Detect armv7 but without the CPU features Rust needs in that build,
     # and fall back to arm.
     # See https://github.com/rust-lang/rustup.rs/issues/587.
-    if [ $_ostype = "unknown-linux-gnueabihf" -a $_cputype = armv7 ]; then
+    if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
         if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
             # At least one processor does not have NEON.
-            local _cputype=arm
+            _cputype=arm
         fi
     fi
 
-    local _arch="$_cputype-$_ostype"
+    _arch="${_cputype}-${_ostype}"
```

Comment:
I think this converts `_arch` to a global.

Reply:
Thanks for reviewing @ehuss. Because we have defined `_arch` as local variable, we don't need to
use `local` keyword when assigning after that. You could check that with this script:
```sh
func ()
{
  local loc_var=23       # Declared as local variable.
  echo                   # Uses the 'local' builtin.
  echo "\"loc_var\" in function = $loc_var"
  global_var=999         # Not declared as local.
                         # Therefore, defaults to global. 
  echo "\"global_var\" in function = $global_var"
}  

func

# Now, to see if local variable "loc_var" exists outside the function.

echo
echo "\"loc_var\" outside function = $loc_var"
                                      # $loc_var outside function = 
                                      # No, $loc_var not visible globally.
echo "\"global_var\" outside function = $global_var"
                                      # $global_var outside function = 999
                                      # $global_var is visible globally.
echo				      

exit 0
#  In contrast to C, a Bash variable declared inside a function
#+ is local ONLY if declared as such.
```
Result:
```commandline
$ sh  ./local.sh 

"loc_var" in function = 23
"global_var" in function = 999

"loc_var" outside function = 
"global_var" outside function = 999
```

Reply:
> Because we have defined `_arch` as local variable

`_arch` is only defined as local in `main`.  It is assigned in `get_architecture`.  Using your example:

```sh
func ()
{
  local loc_var=23       # Declared as local variable.
  echo                   # Uses the 'local' builtin.
  echo "\"loc_var\" in function = $loc_var"
}

func2 ()
{
  loc_var=42
  echo                   # Uses the 'local' builtin.
  echo "\"loc_var\" in function2 = $loc_var"
}

func
echo "\"loc_var\" outside function = $loc_var"
func2
echo "\"loc_var\" outside function = $loc_var"
```

Reply:
I hope resolve this issue by #1604.